server/autolinkplugin/command.go

```diff
@@ -329,6 +350,31 @@ func parseLinkRef(p *Plugin, requireUnique bool, args ...string) ([]autolink.Aut
 	return links, found, nil
 }
 
+func searchLinkRefByTemplateOrPattern(p *Plugin, header *model.CommandArgs, args ...string) ([]autolink.Autolink, []int, error) {
+	links := p.getConfig().Sorted().Links
+	if len(args) == 1 {
+		return links, nil, nil
+	}
+
+	restOfCommand := header.Command[len(autolinkCommand):]
+	restOfCommand = restOfCommand[strings.Index(restOfCommand, args[0])+len(args[0]):]
+	value := strings.TrimSpace(restOfCommand)
+
+	found := []int{}
+	for i, l := range links {
```

Comment:
Outside the scope of this PR, but after just a quick look, I don't see why we need to return the array of indexes. Couldn't the caller just iterate over the returned links? @levb

Reply:
I can't remember off the top of my head, but does it have to do with the indices listed with `/autolink list`?